hive/hive-export-import-metastore.md

```diff
@@ -23,73 +23,98 @@ ms.author: dkakadia
-            else
-                # not partitioned
-                beeline -u "jdbc:hive2://localhost:10001/$d;transportMode=http" --showHeader=false --silent=true --outputformat=tsv2 -e "create external table ext_$t like $t TBLPROPERTIES ('transactional'='false');";
-                staging_ddl=`beeline -u "jdbc:hive2://localhost:10001/$d;transportMode=http" --showHeader=false --silent=true --outputformat=tsv2 -e "show create table ext_$t;"`;
-                dir=$(echo $staging_ddl | grep -io " LOCATION .*" | grep -m1 -o "'.*" | sed "s/'[^-]*//2g" | cut -c2-);
-
-                beeline -u "jdbc:hive2://localhost:10001/$d;transportMode=http" --showHeader=false --silent=true --outputformat=tsv2 -e "insert into ext_$t select * from $t;";
-                echo "load data inpath '$dir' into table $t;" >> alltables.hql;
-                beeline -u "jdbc:hive2://localhost:10001/$d;transportMode=http" --showHeader=false --silent=true --outputformat=tsv2 -e "drop table ext_$t";
-            fi
-        fi
-        echo "$ddl" | grep -q "PARTITIONED\s*BY" && echo "MSCK REPAIR TABLE $t;" >> alltables.hql;
-    done;
-done
-~~~
-
-This will generate a file named `alltables.hql`.
-
-3) Copy the file `alltables.hql` to the new HDInsight cluster and run the following command:
-
-~~~
-beeline -u "jdbc:hive2://localhost:10001/;transportMode=http" -f alltables.hql
-~~~
-
-Note: This assumes that data paths on new cluster are same as on old. If not, you can manually edit the generated  
-`alltables.hql`  file to reflect any changes. *For ACID tables, a new copy of the data will be created*
-
-Note: This script also assumes that once the script is complete, the old cluster will **not** be used any longer 
-
-#### Further Reading:
-
-1) [Connect to HDInsight Cluster using SSH](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-linux-use-ssh-unix)
+Need to export Hive metastore and import it on another HDInsight cluster.
+
+* See [apache-hive-migrate-workloads](https://docs.microsoft.com/en-us/azure/hdinsight/interactive-query/apache-hive-migrate-workloads) for migration of an external metastore, or for upgrading Hive from HDInsight 3.6 to 4.0.
+
+* This current article gives instructions on using a script to export/import contents of an internal Hive metastore.
+
+* Import works only if the destination cluster shares the same Storage Account as the source cluster.
+
+> [!NOTE]
+>
+> * The script supports copying of tables and partitions. For HDInsight 4.0, it additionally covers constraints, views, and materialized views. Other metadata objects, like UDFs, must be copied manually.
+>
+> * All managed tables will become transactional if the output HDInsight version is 4.0. Optionally, make the table non-transactional by exporting the data to an external table with the property 'external.table.purge'='true'. For example,
+>
+>    ```SQL
+>    create table tablename_backup like tablename;
```

Comment:
nit: the syntax highlighting is messed up here in the viewer

Reply:
I don't see it. Do we mean the nested SQL box within the NOTE box? Is there a better style for this?

Reply:
If you mean the stretched out rendering in VSCode, it renders better in the actual docs site.